fn/func.go

```diff
@@ -60,6 +60,18 @@ func FilterMap[T any, K comparable](s map[K]T, f func(T) bool) []T {
 	return output
 }
 
+// FlatMap applies the given mapping function to each element of the given
+// slice and concatenates the results into a new slice.
+func FlatMap[I, O any, S []I](s S, f func(I) []O) []O {
```

Comment:
I think it would be very useful if (after the release is out, in a more quiet minute) to consolidate the `fn` package between `lnd` and `tapd`, then only use one. It happened to me several times that I didn't understand why a `WhenSome()` closure gave me compilation errors and it was just because the wrong `fn` package was being used :sweat_smile:

Reply:
Yeah I also started to have `lfn` vs `fn` at times once I ran into weird compile errors (func doesn't exist, what do you mean, ofc it does!). 

I also spoke with @ProofOfKeags elsewhere about just using a `go.work` in the top level package of lnd for `fn`. Then this way we don't need to do the module dance as much. Not sure how that interacts w/ external packages though.